src/proto/pipeline/client.rs

```diff
@@ -104,10 +104,17 @@ impl<T, E> Task for Client<T, E>
                             let c = self.in_flight.remove(0);
                             c.complete(resp);
                         }
+                        Frame::Error(e) => {
+                            if !self.in_flight.is_empty() {
+                                let c = self.in_flight.remove(0);
```

Comment:
It basically was a `Vec` to get something done. It should definitely not stay a `Vec`